libsolidity/interface/CompilerStack.cpp

```diff
@@ -194,7 +217,30 @@ bool CompilerStack::parse()
 				else
 					noErrors = false;
 
-				m_contracts[contract->name()].contract = contract;
+				if (m_contracts.find(contract->fullyQualifiedName()) != m_contracts.end())
+				{
+					const ContractDefinition* existingContract = m_contracts.find(contract->fullyQualifiedName())->second.contract;
```

Comment:
Sure.  It's safe to use [] by that point.  Also, I appreciate the style nits; y'all just use a different style than I've seen before (though I'm far more a C guy than a C++ guy)